src/components/ResetPassword/index.jsx

```diff
@@ -0,0 +1,95 @@
+/* eslint-disable react/prop-types */
+import React, { useState, useRef } from 'react';
+import { connect } from 'react-redux';
+import { ClipLoader } from 'react-spinners';
+import Modal from '../../views/Modal.js';
+import passwordResetRequest from '../../actions/resetPassword';
+
+
+/**
+ * Form for resetting password
+ * @param {object} props Component props
+ * @param {function} props.passwordResetRequest action creator function
+ */
+function ResetPassword(props) {
+  const [userEmail, setUserEmail] = useState({
+    email: null,
+  });
+
+  const { isLoading } = props;
+
+  const [message, setMessage] = useState(null);
+  const input = useRef(null);
+  
+  const updateInput = event => {
+    setUserEmail({ ...userEmail, [event.target.name]: event.target.value });
+  };
+
+  const clearInputs = () => {
+    input.current.value = null;
+    setUserEmail({ ...userEmail, email: null });
+  };
+
+  const handleSubmit = () => {
+    setMessage(null);
+    props.passwordResetRequest(userEmail.email).then(res => {
+      clearInputs();
+      if (res.code === 400) {
+        setMessage(res.data[0].message);
+      }
+      if (res.code === 200) {
+        setMessage(res.message);
+      }
+    });
+  };
+
+  const validateUserData = (event) => {
+    event.preventDefault();
+    const fieldsArr = Object.entries(userEmail);
+    const errors = [];
+    fieldsArr.forEach(field => {
+      if (!field[1]) {
+        errors.push(`${field[0]} is required.`);
+      }
+    });
+    if (errors.length < 1) {
+      return handleSubmit(event);
+    } else {
+      setMessage(errors);
+    }
+  };
+
+  return (
+    <Modal clearMessage={() => setMessage(null)} title="Recover Password!">
```

Comment:
#### Nitpick
- What happened to the pattern of having views and container component?

Reply:
There are two component involved, the modal is a view while this one is a container